### ISSUE TITLE
`Path` does not set properly inside neovim terminal on windows

### ISSUE THREAD

**Issue by rogtino** [nushell/nushell#12914]
### Describe the bug

I'm currently using nushell as my default shell inside neovim terminal on windows.
path should be `Path` in windows.
But when I open a terminal inside a neovim instance ,nushell warns me that `$env.Path` is not set:
```
Error: nu::shell::name_not_found

  × Name not found
    ╭─[C:\Users\kn1e1\AppData\Roaming\nushell\env.nu:81:23]
 80 │ if $nu.os-info.name == "windows" {
 81 │     $env.Path = ($env.Path | split row (char esep))
    ·                       ──┬─
    ·                         ╰── did you mean 'PATH'?
 82 │     $env.Path = ($env.Path | append '~/.local/bin' |append '~/.cargo/bin')
    ╰────

```
```
❯ echo $nu.os-info.name
windows
```
Path is set properly in a normal terminal.

### How to reproduce

set nushell as default shell in neovim,then open a terminal inside a neovim on windows.

### Expected behavior

PATH should be interpreted as 'Path`

### Configuration

```
❯ version | transpose key value | to md --pretty
| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.93.0                                          |
| major              | 0                                               |
| minor              | 93                                              |
| patch              | 0                                               |
| branch             |                                                 |
| commit_hash        | ca5ee3dba328468b10989f0c85d94a6c0284b767        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)             |
| rust_channel       | 1.77.2-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)             |
| build_time         | 2024-05-01 11:42:46 -07:00                      |
| build_rust_channel | release                                         |
| allocator          | mimalloc                                        |
| features           | default, sqlite, system-clipboard, trash, which |
| installed_plugins  |
```

**Comment by Initsnow** [comment]
This also happens when I run nushell using the msys2 environment

**Comment by ysthakur** [comment]
Looks like this was fixed in https://github.com/nushell/nushell/pull/12701. You should be able to use `$env.Path` in addition to `$env.PATH` now. Can you try upgrading to a newer Nushell version and see if it works?

**Comment by Initsnow** [comment]
> Looks like this was fixed in #12701. You should be able to use `$env.Path` in addition to `$env.PATH` now. Can you try upgrading to a newer Nushell version and see if it works?

It worked.

**Comment by rogtino** [comment]
i don't use windows during this period
since @Initsnow have tested it，I'll close it.
